python/tests/test_tutorial.py

```diff
@@ -107,27 +150,36 @@ def run_simulation(self, vw, num_iterations, users, times_of_day, actions, cost_
             # We negate this so that on the plot instead of minimizing cost, we are maximizing reward
             self.ctr.append(-1*self.cost_sum/i)
 
+            if self.has_aml and i % 1 == 0:
```

Comment:
isn't ` i % 1` always zero?

Reply:
yes it is. I have to remove it but it was useful for debugging at some point. This file is less relevant though maybe spend more time on the cpp unit tests.